### PR TITLE
Feature/WelcomeUserModal

### PR DESCRIPTION
**Pull request by code2244**
# Description

Added backend and frontend functionality to detect if user is the first time login in Together, if it is, a modal with a welcome message will pop up. After user close the message, it will activate a function to delete needsToBeWelcome property from the user schema so the welcome modal doesn't pop up again next time user logs in.

## Type of change

Please select everything applicable. Please, do not delete any lines.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Issue
- [ ] Is this related to a specific issue? Is so please specify: #247 

# Checklist:

- [x] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by code2244** [comment]
Testing this feature I found that if the user doesn't click the button " close" and click outside the WelcomeUserModal box, the modal will close but the property needsToBeWelcome doesn't get deleted and WelcomeUserModal will pop up again the second time the user login or refresh the app.

**Comment by Caleb-Cohen** [comment]
Take a look at this: https://github.com/code2244/Together/pull/1

File: client/src/contexts/AuthContext/useProvideAuth.js [hunk 1]
```diff
@@ -29,12 +29,28 @@ const useProvideAuth = () => {
...
+    return user.needsToBeWelcome;
+  }
+
+  // Delete property of needsToBeWelcome 
+  const deleteNeedsToBeWelcome = () => {
+    const updateUser = { ...user };
+    delete updateUser.needsToBeWelcome;
+    setUser(updateUser)
+    DataService.deleteNeedsToBeWelcome();
+    console.log(user)
```
Comment (Caleb-Cohen) [inline client/src/contexts/AuthContext/useProvideAuth.js:43]: Do we still need this?

File: server/config/passport.js [hunk 2]
```diff
@@ -43,13 +43,16 @@ module.exports = function (passport) {
...
+              needsToBeWelcome: true,
             });
+            console.log(user)
             return cb(null, user);
           } else {
             // it user already exists, update display name and avatar
             user.displayName = displayName;
             user.avatar = profile.avatar;
             const updatedUser = await user.save();
+            console.log(updatedUser)
```
Comment (Caleb-Cohen) [inline server/config/passport.js:55]: this as well!

File: server/config/passport.js [hunk 3]
```diff
@@ -43,13 +43,16 @@ module.exports = function (passport) {
               avatar: profile.avatar,
               socials: [],
               bio: "",
+              needsToBeWelcome: true,
             });
+            console.log(user)
```
Comment (Caleb-Cohen) [inline server/config/passport.js:48]: this as well!

File: server/routes/main.js [hunk 4]
```diff
@@ -29,11 +29,20 @@ router.get(
 //The req.user, null, or a string "User is not in 100Devs"
 //Considering a string is truthy this .get distinguishes between the string and req.user
 router.get("/getDisplayName", (req, res) => {
+  console.log(req.user)
```
Comment (Caleb-Cohen) [inline server/routes/main.js:32]: this as well!

**Review by Caleb-Cohen — CHANGES_REQUESTED** [review]

File: client/src/contexts/AuthContext/useProvideAuth.js (hunk 1 above)
Comment (code2244) [reply client/src/contexts/AuthContext/useProvideAuth.js:43]: Yes, that's is the code that is making the delete as soon as deleteNeedsToBeWelcone() is called.

File: server/config/passport.js (hunk 2 above)
Comment (code2244) [reply server/config/passport.js:55]: no.

File: server/config/passport.js (hunk 3 above)
Comment (code2244) [reply server/config/passport.js:48]: no.

File: server/routes/main.js (hunk 4 above)
Comment (code2244) [reply server/routes/main.js:32]: no.

Comment (Caleb-Cohen) [reply server/routes/main.js:32]: Ok cool. I can get it merged once the extraneous console.logs are removed.

**Review by code2244 — COMMENTED** [review]
Deleted console.logs

**Pull request merged**